sdks/python/apache_beam/dataframe/schemas.py

```diff
@@ -146,15 +146,21 @@ class BatchRowsAsDataFrame(beam.PTransform):
   Batching parameters are inherited from
   :class:`~apache_beam.transforms.util.BatchElements`.
   """
-  def __init__(self, *args, **kwargs):
+  def __init__(self, *args, proxy, **kwargs):
```

Comment:
I think this should just generate the proxy from `pcoll.element_type` in expand. I don't know if there's a good way to retrieve it in `to_pcollection` though... it could just be re-generated, or memoized.

This is one reason I was thinking about a pandas typehint. If we had that this transform could annotate the output PC with a typehint containing the proxy and to_pcollection would be able to retrieve it.

Reply:
I think it's preferable to compute it in one place and pass the desired shape around elsewhere. In addition, this opens the possibility that the user may want to manually specify a shape that cannot be inferred (or would be inferred differently, e.g. the user wants a Series of floats and passes a PCollection of float-convertable things). Granted we don't yet expose that to the public API, but it would make sense in the future.

Reply:
Yeah definitely preferable to generate once... what if we make this `proxy=None` for now, with default behavior to generate the proxy. In the future we could consider propagating that proxy to DataframeTransform through a typehint to remove the need for generating it in `to_pcollection`

Reply:
Done.